File: app/src/main/java/farasense/mobile/model/realm/FaraSenseSensor.java
package farasense.mobile.model.realm;

import com.google.gson.annotations.SerializedName;

import java.util.Date;

import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;

public class FaraSenseSensor extends RealmObject {

    @SerializedName("id")
    Integer id;
    @SerializedName("amper")
    Double amper;
    @SerializedName("power")
    Double power;
    @SerializedName("timestamp")
    Date date;
    @PrimaryKey
    @SerializedName("long_timestamp")
    Long timestamp;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    public Long getTimestamp() {
        return timestamp;
    }

    public void setTimestamp(Long timestamp) {
        this.timestamp = timestamp;
    }

    public double getAmper() {
        return amper;
    }

    public void setAmper(double amper) {
        this.amper = amper;
    }

    public double getPower() {
        return power;
    }

    public void setPower(double power) {
        this.power = power;
    }

}

File: app/src/main/java/farasense/mobile/model/realm/FaraSenseSensor.kt
package farasense.mobile.model.realm

import com.google.gson.annotations.SerializedName
import java.util.Date
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey

open class FaraSenseSensor : RealmObject() {

    @PrimaryKey
    @SerializedName("long_timestamp")
    var timestamp: Long? = null
    @SerializedName("id")
    internal var id: Int? = null
    @SerializedName("amper")
    internal var amper: Double? = null
    @SerializedName("power")
    internal var power: Double? = null
    @SerializedName("timestamp")
    lateinit var date: Date

    fun getId(): Int {
        return id!!
    }

    fun setId(id: Int) {
        this.id = id
    }

    fun getAmper(): Double {
        return amper!!
    }

    fun setAmper(amper: Double) {
        this.amper = amper
    }

    fun getPower(): Double {
        return power!!
    }

    fun setPower(power: Double) {
        this.power = power
    }
}
